### PR TITLE
replace errors.* with clues.*

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Mostly find/replace on errors.N and errors.W. Also turns all wrapf into wrap, and removes as many errorf calls as possible.

Might follow up with a linter to enforce this change.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #1970

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/cli/config/config.go [hunk 1]
```diff
@@ -6,10 +6,10 @@ import (
 	"path/filepath"
 	"strings"
 
-	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 
+	"github.com/alcionai/clues"
```
Comment (ashmrtn) [inline src/cli/config/config.go:12]:
```suggestion
	"github.com/alcionai/clues"
	"github.com/spf13/cobra"
	"github.com/spf13/viper"

```

File: src/cli/repo/s3.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ import (
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 
+	"github.com/alcionai/clues"
```
Comment (ashmrtn) [inline src/cli/repo/s3.go:11]: needs fixed

File: src/cli/utils/exchange.go [hunk 3]
```diff
@@ -1,8 +1,7 @@
 package utils
 
 import (
-	"errors"
-
+	"github.com/alcionai/clues"
```
Comment (ashmrtn) [inline src/cli/utils/exchange.go:4]: Almost had this one, though the rest of the imports could use some help
```suggestion
	"github.com/alcionai/clues"
	
```

File: src/internal/kopia/upload.go [hunk 4]
```diff
@@ -757,31 +752,26 @@ func inflateCollectionTree(
...
-				s.FullPath(),
-			)
+			return nil, nil, clues.New("getting tree node").With("collection_fullpath", s.FullPath())
 		}
 
 		// Make sure there's only a single collection adding items for any given
 		// path in the new hierarchy.
 		if node.collection != nil {
-			return nil, nil, errors.Errorf("multiple instances of collection at %s", s.FullPath())
+			return nil, nil, clues.New("multiple instances of collection").With("collection_fullpath", s.FullPath())
```
Comment (ashmrtn) [inline src/internal/kopia/upload.go:774]:
```suggestion
			return nil, nil, clues.New("multiple instances of collection").With("collection_full_path", s.FullPath())
```

File: src/internal/operations/backup_integration_test.go [hunk 5]
```diff
@@ -448,7 +447,7 @@ func toDataLayerPath(
 	case path.OneDriveService:
 		p, err = pb.ToDataLayerOneDrivePath(tenant, user, isItem)
 	default:
-		err = errors.Errorf("unknown service %s", service.String())
+		err = clues.Wrap(clues.New(service.String()), "unknown service")
```
Comment (ashmrtn) [inline src/internal/operations/backup_integration_test.go:450]: maybe another time to use `Sprintf` to avoid unnecessesary `Wrap` calls?

**Review by ashmrtn — COMMENTED** [review]
imports need reordering in many files

Also please do make a lint rule for this either in this PR or another one

File: src/cli/config/account.go [hunk 6]
```diff
@@ -3,9 +3,9 @@ package config
 import (
 	"os"
 
-	"github.com/pkg/errors"
 	"github.com/spf13/viper"
 
+	"github.com/alcionai/clues"
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/cli/config/account.go:8]: /me wailing and gnashing teeth.

File: src/internal/connector/discovery/api/users.go [hunk 7]
```diff
@@ -182,7 +182,7 @@ func (c Users) GetInfo(ctx context.Context, userID string) (*UserInfo, error) {
 func validateUser(item any) (models.Userable, error) {
 	m, ok := item.(models.Userable)
 	if !ok {
-		return nil, clues.Stack(clues.New("unexpected model"), errors.Errorf("%T", item))
+		return nil, clues.Wrap(errors.Errorf("%T", item), "unexpected model")
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/connector/discovery/api/users.go:185]: yeah, sure

File: src/internal/operations/backup_integration_test.go (hunk 5 above)
Comment (ryanfkeepers) [reply src/internal/operations/backup_integration_test.go:450]: sixes to me.  It's a nested pair of funcs either way.

File: src/cli/restore/exchange.go [hunk 8]
```diff
@@ -5,6 +5,7 @@ import (
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 
+	"github.com/alcionai/clues"
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/cli/restore/exchange.go]: We still call `errors.Is` and `errors.As`.

File: src/cli/restore/onedrive.go [hunk 9]
```diff
@@ -5,6 +5,7 @@ import (
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 
+	"github.com/alcionai/clues"
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/cli/restore/onedrive.go]: 🐸

File: src/cli/restore/sharepoint.go [hunk 10]
```diff
@@ -5,6 +5,7 @@ import (
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 
+	"github.com/alcionai/clues"
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/cli/restore/sharepoint.go]: 🙏

File: src/internal/kopia/upload.go [hunk 11]
```diff
@@ -827,16 +817,20 @@ func traverseBaseDir(
 	dir fs.Directory,
 	roots map[string]*treeMap,
 ) error {
+	ctx = clues.Add(ctx,
+		"old_dir_path", oldDirPath,
+		"new_dir_path", newDirPath)
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/kopia/upload.go:822]: can I change the varname to `expectedDirPath`?  Seems like we'd want both to match.

[the remark this answers is not in the document]
Comment (ashmrtn) [reply src/internal/kopia/upload.go]: sure

**Review by ashmrtn — APPROVED** [review]

**Comment by ryanfkeepers** [comment]
note: this depends on https://github.com/alcionai/corso/pull/2937 to pass

**Pull request merged**